include/pybind11/pytypes.h

```diff
@@ -1216,6 +1217,11 @@ class tuple : public object {
     explicit tuple(size_t size = 0) : object(PyTuple_New((ssize_t) size), stolen_t{}) {
         if (!m_ptr) pybind11_fail("Could not allocate tuple object!");
     }
+    explicit tuple(std::initializer_list<object> init_list) : tuple(init_list.size()) {
+        size_t index {0};
+        for (const pybind11::object& item : init_list)
+            detail::tuple_accessor(*this, index++) = item;
+    }
```

Comment:
With this, we could think about deprecating setting elements of a tuple, since tuples are supposed to be immutable once constructed?

Reply:
Makes sense to me.

Reply:
Nothing for this PR ofc :-)